src/Workspaces/Core/Portable/Workspace/CommandLineProject.cs

```diff
@@ -187,22 +128,54 @@ public static ProjectInfo CreateProjectInfo(string projectName, string language,
                     // TODO (https://github.com/dotnet/roslyn/issues/4967): 
                     .WithMetadataReferenceResolver(new WorkspaceMetadataFileReferenceResolver(metadataService, new RelativePathResolver(ImmutableArray<string>.Empty, projectDirectory))),
                 parseOptions: commandLineArguments.ParseOptions,
-                documents: docs,
+                documents: CreateDocuments(commandLineArguments.SourceFiles),
                 projectReferences: null,
                 metadataReferences: boundMetadataReferences,
                 analyzerReferences: boundAnalyzerReferences,
-                additionalDocuments: additionalDocs,
-                analyzerConfigDocuments: null,
+                additionalDocuments: CreateDocuments(commandLineArguments.AdditionalFiles),
+                analyzerConfigDocuments: CreateDocuments(commandLineArguments.AnalyzerConfigPaths.SelectAsArray(p => new CommandLineSourceFile(p, isScript: false))),
                 hostObjectType: null);
 
             return projectInfo;
+
+            IList<DocumentInfo> CreateDocuments(ImmutableArray<CommandLineSourceFile> files)
```

Comment:
Is this just extracting the helper and additionally uses it for analyzerConfigDocuments?  Is there other changes in how those documents are created?

Reply:
@genlu: yeah for some reason it was copied/pasted for additional files. The only change I saw was special handling around regular/script, but since the compiler's inputs will just count additional files as "regular" that wouldn't be a problem in the first place.